pyw3d/objects.py

```diff
@@ -839,10 +941,20 @@ def blend(self):
             #TODO
         blender_object.game.physics_type = 'DYNAMIC'
         blender_object.game.use_ghost = True
-
-        self.apply_material(blender_object)
-        blender_object.layers = [layer == 0 for layer in range(20)]
-
+       
+        
+        if bpy.data.lamps[-1] is not None and bpy.data.lamps[-1].name[0:13] != 'light_object_':
+            blender_lamp = bpy.data.lamps[-1]                    
+            blender_lamp.name = generate_light_object_name(self["name"])
+            self.apply_lamp_color(blender_lamp)
+        
+        else:
+            self.apply_material(blender_object)
```

Comment:
As a note to myself, I think this might be outdated at this point anyway? I should look into it. Not anything for you to worry about.